### ISSUE TITLE
"Command not found" and open_basedir 

### ISSUE THREAD

**Issue by Junker** [psliwa/image-optimizer#30]
get "Command not found" error when use open_basedir restriction. 

cannot use is_executable when open_basedir path defined and executable optimizer outside of that path.

**Comment by psliwa** [comment]
The same as: https://github.com/psliwa/image-optimizer/issues/16

The library tries to figure out where binaries are - it doesn't work for every OS and all possible configurations - it is not possible. `symfony/process` library is used for figuring out where the binaries are and it uses `open_basedir`. It you have such restriction, you need to provide paths to binaries explicitly. Here you have info how to do this: https://github.com/psliwa/image-optimizer#configuration (`*_bin` options).

**Comment by Junker** [comment]
I explained where is problem. Problem with function is_executable(Command.php, row 23). if comment that block then everything works perfectly. 
It's not connected to symfony/process!
Please, add some boolean option, something like "check_exec"...

**Comment by Junker** [comment]
"*_bin options" will not help, cuz is_executable anyway will send exception, cuz is_executable cannot check file outside of open_basedir path

**Comment by psliwa** [comment]
Thanks, I will take a look at this, I need reproduce this first and make 100% sure that skipping `is_executable` will solve this problem.

**Comment by nlzet** [comment]
@psliwa @Junker I experienced the same issues regarding this topic. I'll note some things I experienced, if you agree I'll provide a PR to fix the experienced issues.

- The problem is indeed, when open_basedir is enabled the method is_executable will always return false when the executable is not within the given open_basedir paths (The same applies to is_file, file_exists, is_readable, is_writable, ...). 
- Even though the is_executable returns false, the exec() method WILL succeed, because it the exec() method doesn't rely on open_basedir. 

A statement I want to make but I'm not 100% sure of this: I think you should be able to disallow open_basedir to the executable so a PHP script can never read/write/delete the executable or other executables in the same directory, the exec() calls can still succeed (chown/chmod permissions can secure unwanted behaviour for exec calls like delete/edit).

TLDR: I want to conclude this library should not check `is_executable`, but instead just rely on the result/exit code returned by the exec method to support environments that deny file access to the executable but allow execute calls.

Can I make a PR for this behaviour ?

**Comment by psliwa** [comment]
@nlzet Thanks for explanation. Yes, PR is welcome.
